Sources/MuxUploadSDK/Public API/MuxUpload+Photos.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MuxUpload+PhotosKit.swift
+//
+
+import Foundation
+import Photos
+
+extension MuxUpload {
+    /// Asynchronously creates a MuxUpload using a ``PHAsset``
+    ///
+    /// - Parameters:
+    ///    - imageManager: Photos image manager
+    ///    - inputAsset: the upload input asset, its
+    ///    ``PHAsset.mediaType`` needs to be ``PHAssetMediaType.video``
+    ///    - uploadURL: the direct upload URL
+    ///    - transportSettings: upload transport settings
+    ///    - optOutOfEventTracking: opts out of SDK event reporting
+    ///    - standardizationSettings: if enabled the SDK checks
+    ///    if the input file is in a standard input format,
+    ///    attempts local conversion if it is non-standard
+    ///    - completion: receives the initialized MuxUpload
+    ///    when it is ready, receives nil if initialization
+    ///    failed
+    public static func makeByExporting(
+        from imageManager: PHImageManager = .default(),
+        using inputAsset: PHAsset,
+        requestOptions: PHVideoRequestOptions,
```

Comment:
Interesting point, agree about sane defaults. 

How would the SDK handle the client setting `deliveryMode` to [fastFormat](https://developer.apple.com/documentation/photokit/phimagerequestoptionsdeliverymode/fastformat)?

Reply:
Currently the SDK will re-export each time (this is coming in PR 2), so the `PHVIdeoRequestOptions` here only matter to the extent they enable the SDK to get an `AVAsset` as a handle to the direct upload input.

Reply:
I'd be concerned that `fastFormat` might provide a degraded video, especially if the video is on iCloud and not locally available. In that case, it might be fastest to download a scaled-down version of the video. Transcoding on top of that will impact quality even if we don't change the resolution or color info at all.

If a developer wishes to make that trade, they could do so, but I would rather the defaults stress quality over local processing time.

Reply:
> From our SDK's perspective, it's all "processing time." To some extent it doesn't matter if we're exporting from a local source or a remote one. Our defaults would be making a decision for people, and I think we should decide to prioritize quality over processing/upload time _except_ where they request otherwise for eg, cost or data use

It is true that the defaults the SDK chooses are key decisions being made on behalf of SDK clients. 

Stepping back from the quality versus inspection, standardization, and upload time tradeoff, I think initially the SDK should be very opinionated and "do the right thing" on behalf of its users. To that end, the purpose (as I see it) of all the `MuxUpload` initializers is for the SDK client to _identify_ the input or inputs they wish to upload. It is then the SDKs job to take of the rest in these initial iterations.

After we get some feedback and better understand the sorts of tradeoffs customers want levers over we can expose finer grained controls over these operations.

> I'd be concerned that `fastFormat` might provide a degraded video, especially if the video is on iCloud and not locally available. In that case, it might be fastest to download a scaled-down version of the video. Us transcoding on top of that will impact quality even if we don't change the resolution or color info at all.
> 
> If a developer wishes to make that trade they could do so, but I would rather the defaults stress quality over local processing time (including potentially caching a remote video).

I think I agree that stressing quality over local processing time is the right call.

Reply:
> After we get some feedback and better understand the sorts of tradeoffs customers want levers over we can expose finer grained controls over these operations.

I see benefit in providing some coarse-grained levers, such as a simple on/off for all standardization, pricing-related things, and dynamic things we aren't yet equipped to decide like chunk size/retry time. I don't think that the `PHVideoRequestOptions` is one of those things, and we probably don't need to expose it

Reply:
Thanks for mentioning the simple on/off (in my mind I hadn't thought of that as a level though of course it is and by definition among the most coarse ones too).

My thought is that on/off and a parameter for the maximum resolution are all we need to expose as APIs and I've updated the PR accordingly. Let me know if you think there are others that are good candidates for this initial version of the feature.